app/src/utils/utils.ts

```diff
@@ -82,6 +83,14 @@ export function decodeMetadataId(id: string): MetaPtr {
   return { protocol: parts[0], pointer: parts[1] };
 }
 
+// Given a logoPtr, resolve it to a URI that we can fetch
+export function ptrToURI(logoPtr: MetaPtr | undefined | null) {
+  if (!logoPtr) return null;
+  const { protocol, pointer } = logoPtr;
+  if (protocol === 1) return getMetaPtr({ cid: pointer });
+  throw new Error(`Unsupported protocol ID ${protocol}`);
+}
```

Comment:
^ not important but throught on removing ptrToURI and moving this logic to `getMetaPtr`

Reply:
`getMetaPtr` is specific to IPFS and lives in `ipfs.ts`, whereas this `ptrToURI` is a wrapper around that to support any supported protocol. For example saw we add Arweave support with a protocol ID of 2, this method would detect that and call `arweave.getMetaPtr` instead of the current `ipfs.getMetaPtr`